src/panels/filter-panel-react/FilterPanelControl.tsx

```diff
@@ -1,45 +1,214 @@
-import { PanelProps } from '@grafana/data'
-import { HorizontalGroup, Input, Select } from '@grafana/ui'
+import React, { useEffect, useState } from 'react'
+import { MetricFindValue, PanelProps, SelectableValue } from '@grafana/data'
+import { getDataSourceSrv } from "@grafana/runtime"
+import { HorizontalGroup, Input, Select, VerticalGroup } from '@grafana/ui'
 import { FieldDisplay } from 'components/FieldDisplay'
-import { saveFilterPanel } from 'lib/localStorageService'
-import React, { useEffect } from 'react'
+import { loadFilterEditorData, saveFilterEditorData } from 'lib/localStorageService'
 import { FilterControlProps } from './FilterPanelTypes'
+import { useEntities } from '../../hooks/useEntities'
+import { ActiveFilter, FilterEditorData, FilterSelectableValues, useFilterData } from '../../hooks/useFilterData'
 
 export const FilterPanelControl: React.FC<PanelProps<FilterControlProps>> = (props) => {
+    const { getFuncNameFromEntityType } = useEntities()
+    const { getFilterId } = useFilterData()
 
-    const getSelectOptions = (filter) => {
-        // TODO: Replace these fake values with the real ones when we can get them from metricfindquery
-        return props?.options?.filterEditor?.properties?.[filter.attribute.value?.id].values ||
-            [{ value: "1", label: '1' }, { value: "2", label: '2' }]
+    // array of metric values for each filter id
+    const [metricValues, setMetricValues] = useState<Array<[string, MetricFindValue[]]>>([])
+
+    // array of selectable values for each filter id
+    const [selectableValues, setSelectableValues] = useState<FilterSelectableValues[]>([])
+
+    useEffect(() => {
+        const filterEditorData = loadFilterEditorData()
+
+        if (filterEditorData && filterEditorData.selectableValues) {
+            setSelectableValues(filterEditorData.selectableValues)
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    useEffect(() => {
```

Comment:
This useEffect and the getValuesFromDatasource below feel like they could be broken down into smaller chunks.